cmd/state-installer/cmd.go

```diff
@@ -442,3 +431,31 @@ func assertCompatibility() error {
 
 	return nil
 }
+
+func determineLegacyUpdate(stateToolInstalled bool, params *Params) bool {
+	// Detect state tool alongside installer executable
+	installerPath := filepath.Dir(osutils.Executable())
+	packagedStateExe := filepath.Join(installerPath, installation.BinDirName, constants.StateCmd+exeutils.Extension)
+
+	// Detect whether this is a fresh install or an update
+	isUpdate := false
+	switch {
+	case (params.sourceInstaller == "install.sh" || params.sourceInstaller == "install.ps1") && fileutils.FileExists(packagedStateExe):
+		logging.Debug("Not using update flow as installing via " + params.sourceInstaller)
+		params.sourcePath = installerPath
+	case params.force:
+		// When ran with `--force` we always use the install UX
+		logging.Debug("Not using update flow as --force was passed")
+	case params.sourcePath == "" && fileutils.FileExists(packagedStateExe):
+		// Facilitate older versions of state tool which do not invoke the installer with `--source-path`
+		logging.Debug("Using update flow as installer is alongside payload")
+		isUpdate = true
+		params.sourcePath = installerPath
+	case stateToolInstalled:
+		// This should trigger AFTER the check above where sourcePath is defined
+		logging.Debug("Using update flow as state tool is already installed")
+		isUpdate = true
+	}
+
+	return isUpdate
+}
```

Comment:
If we've determined setting sourcePath is no longer need then we should be able to drop it here as well.